cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -86,6 +87,10 @@ public ExecutionMetrics[] retrieveMetrics(ProgramRunId runId) throws MetricRetri
     long startTime = System.currentTimeMillis();
     String runIdStr = runId.getRun();
     String sparkHistoricBaseURL = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_HOST);
+    if (Strings.isNullOrEmpty(sparkHistoricBaseURL)) {
+      emitMetrics(runId, startTime, false);
+      throw new MetricRetrievalException("Spark metrics host is missing.");
```

Comment:
Would this cause App Fabric to restart?

Reply:
I don't think do. This causes failure of SparkProgramStatusMetricsProvider.retrieveMetrics().

Reply:
Why throw exception? We can just log and return an empty `ExecutionMetrics[]`?

Reply:
Done